### PR TITLE
Fix #8

### LINKED ISSUE #8: *middleware.compressResponseWriter is not http.CloseNotifier:
Hello!!!
Try using go-see with go-chi as a router by changing:

    http.Handle("/events/", s)

by:

    r.Mount("/events/", s)

but I get the following error:

    panic: interface conversion: *middleware.compressResponseWriter is not http.CloseNotifier: missing method CloseNotify
    ...
    .../github.com/alexandrevicenzi/go-sse/sse.go:82 +0x4c4

I hope you can help me

### PR DESCRIPTION
**Pull request by basebandit**
Replace deprecated [CloseNotifier](https://golang.org/src/net/http/server.go?s=7899:8020#L208 ) with Context.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1 +1,5 @@
 module github.com/alexandrevicenzi/go-sse
+
+go 1.12
+
+require github.com/go-chi/chi v4.0.2+incompatible
```
Comment (alexandrevicenzi) [inline go.mod]: This should not be here as it's not a dependency. Maybe examples needs to be renamed to _examples.

File: go.mod [hunk 2]
```diff
@@ -1 +1,5 @@
 module github.com/alexandrevicenzi/go-sse
+
+go 1.12
```
Comment (alexandrevicenzi) [inline go.mod]: go 1.11 should be here I think, otherwise may not be compatible.

File: examples/custom.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package main
```
Comment (alexandrevicenzi) [inline examples/custom.go]: this file should be called chi.go or something like that, so it's clear what is being used.

Comment (basebandit) [reply examples/custom.go]: Yeah I agree.Let me rename it.

Comment (basebandit) [reply examples/custom.go]: I think removing the custom file would me much appropriate,and just leave the initial examples untouched.

File: go.mod (hunk 2 above)
Comment (basebandit) [reply go.mod]: Adding support for go 1.12 in travis-ci should work.

File: examples/custom.go (hunk 3 above)
Comment (alexandrevicenzi) [reply examples/custom.go]: I think this example is useful.

Comment (basebandit) [reply examples/custom.go]: I am wondering how to include the go-chi dependency without polluting our go.mod

File: go.mod (hunk 1 above)
Comment (basebandit) [reply go.mod]: Renamed examples to _examples.

Comment (basebandit) [reply go.mod]: I think you should consider cleaning test cache on travis before running them.

File: go.mod [hunk 4]
```diff
@@ -1 +1,3 @@
...
```
Comment (alexandrevicenzi) [inline go.mod]: Could it be 1.11 here?

Comment (basebandit) [reply go.mod]: It was autogenerated by the go tool.

Comment (alexandrevicenzi) [reply go.mod]: Because you are using 1.12, if you use the docker image inside it's 1.11 and would not change anything.

Comment (basebandit) [reply go.mod]: Oh sure,I had forgot bout the docker.Let me change it.

Comment (basebandit) [reply go.mod]: On hindsight I could just update the docker-compose script to pull a go1.12 image.

**Comment by alexandrevicenzi** [comment]
@basebandit can you add a test for this case? I'm not sure if we have something that covers this change.

**Pull request merged**